fs/background-fetcher/background_fetcher.go

```diff
@@ -0,0 +1,162 @@
+/*
```

Comment:
How about moving that to fs/backgroundfetcher/background_fetcher.go or fs/bakground_fetcher.go?

- Go package names are generally the same as the file hierarchy even that have multiple words (see https://github.com/golang/go/tree/master/src/index/suffixarray)
- https://dave.cheney.net/practical-go/presentations/qcon-china.html#_consider_fewer_larger_packages

Reply:
I think it makes sense to have  a ``backgroundfetcher`` package as we expect to see the fetcher implementation to be extended in the future.

Reply:
Renamed package to `backgroundfetcher`. We also have a `spanmanager` package in a `span-manager` directory that we should update in a future PR.